pkg/ipmi/ipmi.go

```diff
@@ -173,3 +188,61 @@ func (i *IPMI) ShutoffWatchdog() error {
 
 	return nil
 }
+
+func (i *IPMI) setsysinfo(data *setSystemInfoReq, dataLen uint16) error {
+	req := &req{}
+	req.msg.cmd = _SET_SYSTEM_INFO_PARAMETERS
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.dataLen = dataLen
+	req.msg.data = unsafe.Pointer(data)
+
+	if _, err := i.sendrecv(req); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SetSystemFWVersion sets the provided system firmware version to BMC via IPMI.
+func (i *IPMI) SetSystemFWVersion(version string) error {
+	strlenMax := 62 // Set 62 Bytes (4 sets) as the maximal version string length
+	len := len(version)
+
+	if len == 0 {
+		return fmt.Errorf("Version length is 0")
+	} else if len > strlenMax {
+		len = strlenMax
+	}
+
+	var data setSystemInfoReq
+	var copied int
+	var dataLen uint16
+	data.paramSelector = _SYSTEM_FW_VERSION
+	data.setSelector = 0
+	index := 0
+	for len > 0 {
```

Comment:
Maybe `for len > index` to avoid having to decrease the length in line 240?

Reply:
Updated.